library/core/src/ptr/mod.rs

```diff
@@ -537,7 +537,9 @@ pub unsafe fn drop_in_place<T: ?Sized>(to_drop: *mut T) {
 #[rustc_allow_const_fn_unstable(ptr_metadata)]
 #[rustc_diagnostic_item = "ptr_null"]
 pub const fn null<T: ?Sized + Thin>() -> *const T {
-    from_raw_parts(invalid(0), ())
+    // Use transmute instead casting so that miri knows that the pointer is invalid.
+    // SAFETY: every usize is a valid pointer (on all current platforms).
+    unsafe { mem::transmute(0usize) }
```

Comment:
This change and the similar change in `null_mut` shouldn't be necessary given the changes to `invalid`, which the old implementation was based on.

Reply:
They are, otherwise this function would infinitely recurse, as `invalid` now calls `null` internally.

Reply:
oh... whoops didn't notice that :sweat_smile: